### PR TITLE
Apply Fix All in Solution for IDE0044 (Make Field Readonly)

### PR DESCRIPTION
**Pull request by sharwell**
I applied a single Fix All in Solution for IDE0044 using the default settings for the Make Field Readonly analyzer (#19067). This is the raw/unaltered result.

### CODE REVIEW

File: src/VisualStudio/CSharp/Impl/ProjectSystemShim/HACK_VariantStructure.cs [hunk 1]
```diff
@@ -23,10 +23,10 @@ namespace Microsoft.VisualStudio.LanguageServices.CSharp.ProjectSystemShim
     internal struct HACK_VariantStructure
     {
         [FieldOffset(0)]
-        private short _type;
+        private readonly short _type;
 
         [FieldOffset(8)]
-        private short _booleanValue;
+        private readonly short _booleanValue;
```
Comment (jasonmalinowski) [inline src/VisualStudio/CSharp/Impl/ProjectSystemShim/HACK_VariantStructure.cs]: Note this is used in GetObjectForNativeVariant as called below, which means these fields *are* mutated. The refactoring is probably incorrect here.

File: src/Workspaces/Core/Portable/NamingStyles/NamingStyle.cs [hunk 2]
```diff
@@ -163,34 +163,32 @@ private WordSpanEnumerable GetWordSpans(string name, TextSpan nameSpan)
...
-            {
-                if (DoesCharacterHaveCasing(val[i]) && !char.IsLower(val[i]))
-                {
-                    return false;
-                }
-            }
+        private static readonly Func<string, TextSpan, bool> s_firstCharIsLowerCase = (val, span) => !DoesCharacterHaveCasing(val[span.Start]) || char.IsLower(val[span.Start]);
+        private static readonly Func<string, TextSpan, bool> s_firstCharIsUpperCase = (val, span) => !DoesCharacterHaveCasing(val[span.Start]) || char.IsUpper(val[span.Start]);
+        private static readonly Func<string, TextSpan, bool> s_wordIsAllUpperCase = (val, span) =>
+{
```
Comment (jasonmalinowski) [inline src/Workspaces/Core/Portable/NamingStyles/NamingStyle.cs]: Formatting got mangled here.

File: src/VisualStudio/CSharp/Impl/ProjectSystemShim/HACK_VariantStructure.cs (hunk 1 above)
Comment (jasonmalinowski) [reply src/VisualStudio/CSharp/Impl/ProjectSystemShim/HACK_VariantStructure.cs]: (I presume the Marshal call will happily overwrite the values and ignore 'readonly'. That probably warrants a #pragma suppress.)

File: src/Workspaces/Core/Portable/PatternMatching/PatternMatcher.cs [hunk 3]
```diff
@@ -33,8 +33,7 @@ internal abstract partial class PatternMatcher : IDisposable
         // PERF: Cache the culture's compareInfo to avoid the overhead of asking for them repeatedly in inner loops
         private readonly CompareInfo _compareInfo;
         private readonly TextInfo _textInfo;
-
-        private bool _invalidPattern;
+        private readonly bool _invalidPattern;
```
Comment (sharwell) [inline src/Workspaces/Core/Portable/PatternMatching/PatternMatcher.cs]: :memo: Erroneous diagnostic fixed in f19cceb66030ae14c9f11850b08d3ac9694059bd

File: src/VisualStudio/CSharp/Impl/ProjectSystemShim/HACK_VariantStructure.cs (hunk 1 above)
Comment (sharwell) [reply src/VisualStudio/CSharp/Impl/ProjectSystemShim/HACK_VariantStructure.cs]: ➡️ This goes the other way around. `GetObjectForNativeVariant` is reading from these fields, not writing to them.

File: src/Workspaces/Core/Portable/NamingStyles/NamingStyle.cs (hunk 2 above)
Comment (sharwell) [reply src/Workspaces/Core/Portable/NamingStyles/NamingStyle.cs]: It did. Annoyingly, it happens in the IDE but so far doesn't happen in the unit tests. Investigating...

Comment (sharwell) [reply src/Workspaces/Core/Portable/NamingStyles/NamingStyle.cs]: ➡️ Formatting issues are now resolved, including these examples as well as a few others where whitespace changed unnecessarily.

**Pull request closed without merging**